### PR TITLE
Migrate to kuberetes-release-dev

### PR DESCRIPTION
**Pull request by zmerlynn**
[Analytics]()

This turns off the kubernetes-release-dev mirror. #28193 should have shifted all the references over, this should ensure that it's true.

Along the way: Modify it so that the equal bucket and mirror is a noop for things like https://github.com/kubernetes/test-infra/pull/244.

### CODE REVIEW

**Comment by zmerlynn** [comment]
@ixdy is OOO, @spxr, care to look? Or @david-mcmahon?

File: build/common.sh [hunk 1]
```diff
@@ -1228,7 +1228,7 @@ function kube::release::gcs::copy_release_artifacts() {
 
   gsutil ls -lhr "${gcs_destination}" || return 1
 
-  if [[ -n "${KUBE_GCS_RELEASE_BUCKET_MIRROR:-}" ]]; then
+  if [[ "${KUBE_GCS_RELEASE_BUCKET_MIRROR:-}" != "${KUBE_GCS_RELEASE_BUCKET}" ]]; then
```
Comment (spxtr) [inline build/common.sh]: If `KUBE_GCS_RELEASE_BUCKET_MIRROR` is empty but `KUBE_GCS_RELEASE_BUCKET` is not will this break?

Comment (zmerlynn) [inline build/common.sh]: Yes. I knew I meant to leave the `-n` in as well. Jussec.

**Comment by zmerlynn** [comment]
PTAL

**Comment by spxtr** [comment]
This looks fine to me.

**Comment by zmerlynn** [comment]
cc @roberthbailey .. I'll release the do-not-merge when the GKE side has been finished.

**Comment by david-mcmahon** [comment]
LGTM

**Comment by zmerlynn** [comment]
I'm self-labeling this after enough LGTM-y comments. :)

**Comment by zmerlynn** [comment]
Approving cherrypick based on prior discussions. Someone can rescind if they object.

**Pull request merged**